mppsolar/protocols/abstractprotocol.py

```diff
@@ -331,6 +332,25 @@ def decode(self, response, command) -> dict:
                         result = 0
                     msgs[key] = [float(result) / 10, resp_format[2]]
                 # eg. ['option', 'Output source priority', ['Utility first', 'Solar first', 'SBU first']],
+
+                #for any simple value change
+                elif re.search("int:r[/*+-]\d+", resp_format[0]) != None:
```

Comment:
should not need this - i'll have a look and see why this is required

Reply:
I could not find a way how to multiply or divide recieved value, so I added this. Similar to 10int, which is already in place.

Reply:
Ok I find it, but I do not know how to format the response_type to get there.

Reply:
I would advice to add this to default in abstract protocol. It will solve the problem and still use function already in use.
```
elif ':' in resp_format[0] :
                    
                    processed_responses = self.process_response(
                        data_name=resp_format[1],
                        raw_value=result,
                        data_units=resp_format[2],
                        data_type=resp_format[0]
                    )

                    msgs[key] = [processed_responses[0][1], data_units]
```

Reply:
OK I see now - you are using the default type processing (which will be removed in future) 
If you change the type to `"type": "SEQUENTIAL",` for each command definition the `:r/10`  template processing will work without changes to abstractprotocol

Reply:
Yes, but sequential can process only numbers not string. What should the response_type mean? There aren't much basic datatypes outside default.

Reply:
sequential processes strings - what problem do you get?

Reply:
File "/home/orangepi/mpp-solar/mppsolar/devices/device.py", line 115, in run_command
    decoded_response = self._protocol.decode(raw_response, command)
  File "/home/orangepi/mpp-solar/mppsolar/protocols/abstractprotocol.py", line 563, in decode
    processed_responses = self.process_response(
  File "/home/orangepi/mpp-solar/mppsolar/protocols/abstractprotocol.py", line 203, in process_response
    r = eval(format_string)
  File "<string>", line 1, in <module>
NameError: name 'string' is not defined

Reply:
try `str`

Reply:
str will return byte value like this:
battery_piece_number              b'04'                 ea
but no error

Reply:
wouldnt int work there?

Reply:
OK, this works, thanks